zopkio/test_recipes.py

```diff
@@ -0,0 +1,89 @@
+# specific language governing permissions and limitations
+# under the License.
+
+from datetime import datetime
+from datetime import time as dtime
+from dateutil import parser
+import os
+import threading
+import random
+import logging
+import time
+
+import zopkio.runtime as runtime
+
+def test_kill_random_deployer_process(deployer_list, deployer_process_dict, deployer_restart_func_dict=None, deployer_verify_recovery_dict=None,deployer_timeout_dict=None):
+  """
+  A test recipe to select a random deployer and kill one of its process.If no process is present in list then all its process will be killed
+  :param deployer_list: List of deployer from which we want to select one and kills its process
+  :param deployer_process_dict: dictionary with list of process ids per deployer from which we need to kill one randomly.
+  :param deployer_restart_func_dict : dictionary with  key as deployer and value as dict of restart commands if any for list of process.
+  :param deployer_verify_recovery_dict: dictionary with  key as deployer and value as verify function per process to verify  if the system was restored correctly.
+  :param deployer_timeout_dict: dictionary with  key as deployer and value as optional timeout parameter in seconds per process specified has dictionary.
+  """
+  if (deployer_list is None):
+    assert " No deployer specified in test_kill_random_deployer_process"
+
+  kill_deployer = random.choice(deployer_list)
+
+  kill_deployer_processes = deployer_process_dict.get(kill_deployer,kill_deployer.get_processes())
+
+  test_kill_recovery(kill_deployer, kill_deployer_processes, deployer_restart_func_dict.get(kill_deployer, None),
+                     deployer_verify_recovery_dict.get(kill_deployer, None), deployer_timeout_dict.get(kill_deployer, None))
+
+
+def test_kill_recovery(kill_deployer, kill_deployer_processes, restart_func_dict=None, verify_recovery_dict=None, timeout_dict=None):
+  """
+  A test recipe to kill a component in distributed system and verify system recovery once its brought back.
+  :param kill_deployer: The deployer one of whose process needs to be killed
+  :param kill_deployer_processes: list of process ids from which we need to kill one randomly.
+  :param restart_func_dict : dictionary with restart commands if any for list of process.None will assume process starts on own
+  :param verify_recovery_dict: Verify function per process to verify  if the system was restored correctly.Wait till this function returns true
+  :param timeout_dict: optional timeout parameter in seconds per process specified has dictionary. If the system did not recover within this time then error out
+  """
+  if (kill_deployer is None):
+    assert "test_kill_recovery called without any deployer"
+
+  if (kill_deployer_processes is None):
+    assert "No process specified for the deployer"
+
+  #Stop the component. If there is a stop command use it else will evoke kill on the process
```

Comment:
this comment is misleading as if just invokes stop regardless and it is the deployer that will evoke kill if no stop command is given